app/controllers/stripe_controller.rb

```diff
@@ -0,0 +1,47 @@
+class StripeController < ApplicationController
+  protect_from_forgery except: :webhook
+
+  def webhook
+    webhook_secret = ENV["STRIPE_WEBHOOK_SIGNING_KEY"]
+    if webhook_secret.present?
+      sig_header = request.env["HTTP_STRIPE_SIGNATURE"]
+      payload = request.body.read
+
+      begin
+        @event = Stripe::Webhook.construct_event(payload, sig_header, webhook_secret)
+
+        event_data = get_event_data
+        metadata = get_metadata
+        case @event.type
+        when "payment_intent.succeeded"
+          order = Order.find_by(id: metadata.order_id)
+          if order.status = "pending"
```

Comment:
Do we need to check whether `order` is `nil`?

Reply:
Sure. We should do it